tests/ludwig/schema/test_config_object.py

```diff
@@ -0,0 +1,373 @@
+        ],
+        "output_features": [
+            {
+                "name": "number_output_feature",
+                "type": "number",
+            },
+        ],
+    }
+
+    config_object.update_with_dict(temp_config)
+
+    assert config_object.input_features.text_feature.encoder.max_sequence_length == 10
+
+
+@pytest.mark.parametrize("load_format", ["dict", "yaml"])
+def test_constructors(load_format):
+    config = {
+        "input_features": [
+            {"name": "text_feature", "type": "text", "encoder": {"type": "parallel_cnn", "max_sequence_length": 10}},
+        ],
+        "output_features": [
+            {
+                "name": "number_output_feature",
+                "type": "number",
+            },
+        ],
+    }
+
+    if load_format == "dict":
+        config_obj = ModelConfig.from_dict(config)
+
+    if load_format == "yaml":
+        with TemporaryDirectory() as tmpdir:
+            file_path = os.path.join(tmpdir, "test.yaml")
+            with open(file_path, "w") as file:
+                yaml.dump(config, file)
+
+            config_obj = ModelConfig.from_yaml(file_path)
+
+    assert hasattr(config_obj, INPUT_FEATURES)
+    assert hasattr(config_obj, OUTPUT_FEATURES)
+    assert hasattr(config_obj, PREPROCESSING)
+    assert hasattr(config_obj, TRAINER)
+    assert hasattr(config_obj, COMBINER)
+    assert hasattr(config_obj, DEFAULTS)
+    assert hasattr(config_obj, HYPEROPT)
+
+
+def test_feature_enabling_disabling():
+    config = {
+        "input_features": [{"name": "text_feature", "type": "text"}, {"name": "category_feature", "type": "number"}],
+        "output_features": [
+            {
+                "name": "number_output_feature",
+                "type": "number",
+            },
+        ],
+    }
+
+    config_obj = ModelConfig.from_dict(config)
+
+    assert config_obj.input_features.text_feature.active
+    assert config_obj.input_features.category_feature.active
+
+    config_obj.input_features.text_feature.disable()
+
+    assert not config_obj.input_features.text_feature.active
+
+
+def test_sequence_combiner():
+    config = {
+        "input_features": [{"name": "text_feature", "type": "text"}],
+        "output_features": [{"name": "number_output_feature", "type": "number"}],
+        "combiner": {"type": "sequence", "encoder": {"type": "rnn"}},
+    }
+
+    config_obj = ModelConfig.from_dict(config)
+
+    assert config_obj.combiner.type == "sequence"
+    assert config_obj.combiner.encoder.type == "rnn"
+    assert config_obj.combiner.encoder.cell_type == "rnn"
+
+
+@pytest.mark.parametrize(
+    "session",
+    [
+        {"sess_id": 0, "encoder": "parallel_cnn", "loss": {"type": "mean_squared_error"}},
+        {"sess_id": 1, "encoder": "cnnrnn", "loss": {"type": "mean_absolute_error"}},
+        {"sess_id": 2, "encoder": "parallel_cnn", "loss": {"type": "mean_absolute_error"}},
+    ],
+)
+def test_shared_state(session):
```

Comment:
So this test is a very tricky part of this PR but very necessary. It is better to explain over a call since it's quite complicated, but the short explanation is that when dataclasses (which we use throughout the schema to generate the full schema) have class attributes set to mutable defaults, those mutable defaults can be altered like class attributes. Thing is, we can't set these as instance attributes for the schema because marshmallow uses the dataclasses to generate the json schema. So what happens here is that these classes will get modified across test runs and cause tests to fail when run in a suite, but they will pass when run individually. It was a really tricky thing to nail down, but we solved it by adding deep copies in strategic places to just get the config classes with instance attributes initialized. This test makes sure that the shared state across test cases is handled appropriately. Here is a link that describes the python dataclass behavior that causes this issue: https://docs.python.org/3/library/dataclasses.html#mutable-default-values